### DISCUSSION TITLE
Local login

### DISCUSSION THREAD

**Discussion by jayhen** [sindrebroch/ha-flexit#27] [Ideas]
Hi,
Great module, thanks. I'm not sure if you're actively developing but is it possible to log in locally instead of via the Flexit servers? I assume this would avoid the API call limit, responses are much faster (att least in the app) and i guess reduces potential problems with internet connectivity. In the Flexit Go app this is achieved by signing out, then Find Product, select the product, and then enter 'local' as the password for a standard user or '1000' for advanced (installer) mode.
Cheers.

**Comment by sindrebroch** [comment]
Hi. Thanks for you suggestion. 

Yes, I started looking into this, but the issue is that local connection uses a different protocol, BACnet. It is pretty difficult to develop against, and I have no experience in it. Not a lot of info on the topic on the internet, as it is used more in industrial settings. 

I have been looking into a python library BACpypes, which is in active development, and I hope that it will be a piece of the puzzle. 

I don't think there are any other HA-integrations that uses BACnet, which means I can't find any practical examples. 

Will see if I have any successful attempts in the future. I agree that this would be a great improvement in all manners.

**Comment by jayhen** [comment]
Hi, looks challenging. This is not HA specific, so not sure if it helps but I came across a recent video describing a tool to connect to and inspect objects on the BACnet device and a different python module to read BACnet properties: https://youtu.be/TyEXDnjBsD8

It would be amazing to have control within the local network, but already fantastic to have any control from HA :)
